store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -309,6 +300,17 @@ public boolean load() {
      */
     @Override
     public void start() throws Exception {
+        if (!messageStoreConfig.isEnableDLegerCommitLog() && !this.messageStoreConfig.isDuplicationEnable()) {
+            this.haService.init(this);
+        }
+
+        if (messageStoreConfig.isTransientStorePoolEnable()) {
+            this.transientStorePool.init();
+        }
```

Comment:
Why put the init operation in the start function?

Reply:
It makes more sense for init method to be called in start function, instead of being inited in the constructor of MessageStore.